jupyter_client/tests/test_multikernelmanager.py

```diff
@@ -29,6 +29,14 @@
 TIMEOUT = 30
 
 
+async def now(awaitable):
+    """Use this function ensure that this awaitable
+    happens before other awaitables defined after it.
+    """
+    (out,) = await asyncio.gather(awaitable)
```

Comment:
I guess I don't understand why this method is necessary - especially for a single-item gather.  Isn't awaiting the awaitable sufficient to prevent the execution of follow-on awaitables in the same execution path?

Reply:
Great question, @kevin-bates. 

That's what's supposed to happen; however, I'm seeing some weird behavior that *I think* is coming from the `gen_test` decorator. Basically, I'm not seeing the `await`s happening in the order they are defined.

Weirdly, if I replace the async/await syntax with the old `yield` syntax, these tests work fine (no `now` needed). Again, this suggest some weird interference happening from Tornado's async testing. I'm not sure how to fix this.

Reply:
By enforcing a `gather` call in each await statement, everything happens in the order they are called. Super strange!

Reply:
Thanks Zach.  That is interesting.

Reply:
Looks like nested coroutines can be a bit difficult in unit tests. It appears that inner coroutines never get scheduled even when the outer coroutine is `await`ed. Adding `ensure_future` in multiple places in the tests ensures that these coroutines get scheduled and tests pass again.